spec/nmt.md

```diff
@@ -0,0 +1,303 @@
+# Introduction
+
+Namespaced Merkle Tree, NMT for short, is one of the core components of Celestia blockchain.
+Transactions in Celestia are associated with a namespace ID which signifies the application they belong to.  
+Nodes interested in a specific application only need to download transactions of a certain namespace ID. 
+The Namespaced  Merkle Tree (NMT) was introduced in the [LazyLedger article](https://arxiv.org/abs/1905.09274) to organize transactions in Celestia blocks based on their namespace IDs. 
+The NMT allows for efficient and verifiable queries of application-specific transactions by accessing only the block header, which contains the NMT root.
+This specification explains the NMT data structure and provides an overview of its current implementation in the repository.
```

Comment:
[non-blocking question]

are we purposefully using a new line on line 4? if so, do we want to add another new line for formatting?

Reply:
Good catch! the newline is not intentional on that line, fixed it now https://github.com/celestiaorg/nmt/pull/101/commits/440b7623b1fcd5cd93aa57c6dbcadc714abf0207